components/tabs/tab_panel.vue

```diff
@@ -80,7 +80,33 @@ export default {
   },
 
   async mounted () {
-    this.isFirstElementFocusable = !!(await this.getFirstFocusableElement(this.$el));
+    const firstFocusableElement = await this.getFirstFocusableElement(this.$el);
+
+    if (!firstFocusableElement) {
+      this.isFirstElementFocusable = false;
+    } else {
+      // If the first focusable element isn't the first element in the panel,
+      // then we need to set the panel tabindex to 0.
+      // See notes in https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/
+      this.isFirstElementFocusable = this.isFirstElementOfPanel(firstFocusableElement);
+    }
+  },
+
+  methods: {
+    isFirstElementOfPanel (element) {
+      let current = element;
+      let isFirstElement = true;
+
+      while (current) {
+        if (current.previousElementSibling !== null) {
+          isFirstElement = false;
+          break;
+        }
+        current = current.parentNode !== this.$el ? current.parentNode : null;
+      }
```

Comment:
Do we even need to loop this? If there's any previous sibling before the firstFocusableElement then we know right away it's not the first element. I think just this would work right?

```js
isFirstElementOfPanel (element) {
  return element.previousElementSibling !== null
}

Reply:
Yeah that should work for the siblings of the element's parent but with the loop we also check for other elements that aren't in that node. For example:

```html
<dt-panel>
  <h2>Content</h2>
  <div id="container>
    <button>button</button>
  </div>
<dt-panel>
```

`element.previousElementSibling` will only check inside the `div#container` but the first element in the panel is the `h2`.

Reply:
Hmm yep I see. This is good then.